lib/model/rwfolder.go

```diff
@@ -192,65 +186,65 @@ func (p *rwFolder) Serve() {
 
 	for {
 		select {
-		case <-p.stop:
+		case <-f.stop:
 			return
 
-		case <-p.remoteIndex:
+		case <-f.remoteIndex:
 			prevVer = 0
-			p.pullTimer.Reset(0)
-			l.Debugln(p, "remote index updated, rescheduling pull")
+			f.pullTimer.Reset(0)
+			l.Debugln(f, "remote index updated, rescheduling pull")
 
-		case <-p.pullTimer.C:
+		case <-f.pullTimer.C:
 			if !initialScanCompleted {
-				l.Debugln(p, "skip (initial)")
-				p.pullTimer.Reset(p.sleep)
+				l.Debugln(f, "skip (initial)")
+				f.pullTimer.Reset(f.sleep)
 				continue
 			}
 
-			p.model.fmut.RLock()
-			curIgnores := p.model.folderIgnores[p.folder]
-			p.model.fmut.RUnlock()
+			f.model.fmut.RLock()
```

Comment:
Similarly, this reaching out into the `model` for all kinds of things is an abomination that should be rooted out. Looking at doing that, by moving more scan functionality into the `folderscan` perhaps, would be neat.